Assets/Scripts/Characters/PlayerMovement.cs

```diff
@@ -53,37 +89,87 @@ private void FixedUpdate()
             _animator.SetBool(IsJumpingID, _isJumping);
         }
 
-        private void OnDisable()
+        #region Server
+
+        [Server]
+        public override void OnStopServer()
         {
             _isJumping = false;
             _animator.SetBool(IsJumpingID, _isJumping);
             _animator.SetBool(IsRunningID, false);
         }
 
-        private void HandleMovement(float movement)
+        [Command]
+        private void CmdMovementStarted()
         {
-            _body.velocity = new Vector2(movement * speed, _body.velocity.y);
+            ServerMovementStarted?.Invoke(connectionToClient.connectionId);
+        }
 
-            if (movement > 0.01f)
+        [Command]
+        private void CmdMovementEnded()
+        {
+            ServerMovementEnded?.Invoke(connectionToClient.connectionId);
+        }
+
+        [Command]
+        private void CmdJump()
+        {
+            ServerPlayerJumped?.Invoke(connectionToClient.connectionId);
+        }
+
+        #endregion
+
+        #region Client
+
+        public override void OnStartAuthority()
+        {
+            _playerInput.actions["Player/Jump"].performed += ClientHandleJump;
+        }
+
+        public override void OnStopClient()
+        {
+            if (!isClientOnly || !hasAuthority)
             {
-                _body.transform.localScale = Vector3.one;
+                return;
             }
-            else if (movement < -0.01f)
+
+            _playerInput.actions["Player/Jump"].performed -= ClientHandleJump;
+        }
+
+        private void ClientHandleMove(float movement)
+        {
+            if (!enabled)
+            {
+                return;
+            }
+            _body.velocity = new Vector2(movement * speed, _body.velocity.y);
+            Vector2 moveDirection = _body.velocity;
+
+            bool isMoving = Mathf.Abs(moveDirection.x) > 0.01f;
+            if (isMoving)
             {
-                _body.transform.localScale = new Vector3(-1, 1, 1);
+                float angle = Mathf.Atan2(moveDirection.y, moveDirection.x) * Mathf.Rad2Deg;
+                transform.rotation = Quaternion.AngleAxis(angle, Vector3.up);
```

Comment:
The previous method wasn't syncing across the network for some reason. The quaternion rotation causes a weird flipping effect occasionally but this works over the network. We can improve it in the future if it looks too weird when we start getting more permanent art